client/state/reader/feed-search/reducer.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { createReducer, keyedReducer } from 'state/utils';
+
+/**
+ * Internal dependencies
+ */
+import { READER_FEED_SEARCH_RECEIVE } from 'state/action-types';
+
+/**
+ * Tracks mappings between queries --> feed results
+ * Here is what the state tree may look like:
+ * feedsearch: {
+		items: {
+			'wordpress tavern': [ feed1, feed2, ],
+			...
+		},
+	}
+ *
+ * @param  {Array} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Array}        Updated state
+ */
+export const items = keyedReducer( 'query', createReducer( [], {
+	[ READER_FEED_SEARCH_RECEIVE ]: ( state, action ) => state.concat( action.payload )
```

Comment:
looking back at it now, I'm not sure thats a good idea.
The response from the API isn't actually feed objects, its a tiny subset:
```
{
 URL,
 feed_ID || blog_ID,
 title,
 railcar,
}
```

Is there any gain in storing this sort of 'minimal' feed object in the feeds reducer?